### PR TITLE
fix: support mismatching timesteps in ltp delta profile

### PR DESCRIPTION
**Pull request by jsolaas**
Refs: https://equinor-ecalc.atlassian.net/browse/ECALC-64

### CODE REVIEW

File: src/libecalc/presentation/exporter/dto/dtos.py [hunk 1]
```diff
@@ -100,6 +116,18 @@ def check_equal_lengths_of_timeseries(cls, values):
 
         return values
 
+    def fit_to_timesteps(self, timesteps: List[datetime]) -> "TSVPrognosis":
+        return TSVPrognosis(
+            time_steps=TimeSteps(
+                values=timesteps,
+                frequency=self.time_steps.frequency,  # Should this be provided?
```
Comment (jsolaas) [inline src/libecalc/presentation/exporter/dto/dtos.py:123]: What's the frequency here @TeeeJay? Just output frequency used for the run? Is it needed?

Comment (TeeeJay) [reply src/libecalc/presentation/exporter/dto/dtos.py:123]: https://en.wikipedia.org/wiki/What%27s_the_Frequency,_Kenneth%3F

Comment (jsolaas) [reply src/libecalc/presentation/exporter/dto/dtos.py:123]: nice

File: src/libecalc/presentation/exporter/dto/dtos.py [hunk 2]
```diff
@@ -69,6 +69,22 @@ class TimeSeries(EcalcResultBaseModel):
     title: str
     unit: Unit
 
+    def fit_to_timesteps(self, timesteps: List[datetime]) -> "TimeSeries":
+        """
+        Fit TimeSeries to timesteps, in this case it is expected that the provided timesteps is a subset of the current
```
Comment (TeeeJay) [inline src/libecalc/presentation/exporter/dto/dtos.py:74]: I think we should check that is actually is a subset?

File: src/libecalc/presentation/exporter/dto/dtos.py (hunk 1 above)
Comment (TeeeJay) [reply src/libecalc/presentation/exporter/dto/dtos.py:123]: but yes, I think it is just "metainfo" , but possibly used to know what to expect from the timeseries interval. shouldnt be needed now i think, as we can change to whatever we want, irrelevant of original frequency

File: src/libecalc/presentation/exporter/dto/dtos.py [hunk 3]
```diff
@@ -178,12 +206,32 @@ class AssetTSVPrognosis(EcalcResultBaseModel):
 
     tsv_prognoses: Dict[str, TSVPrognosis] = Field(default_factory=dict)
 
+    @property
+    def timesteps(self) -> Set[datetime]:
```
Comment (TeeeJay) [inline src/libecalc/presentation/exporter/dto/dtos.py:210]:
```suggestion
    def common_timesteps(self) -> Set[datetime]:
```
? vs union, if that is suddenly needed (call it sth with intersection?)

**Review by TeeeJay — APPROVED** [review]

File: src/libecalc/presentation/exporter/dto/dtos.py (hunk 3 above)
Comment (jsolaas) [reply src/libecalc/presentation/exporter/dto/dtos.py:210]: Yes, I wasn't sure, as all the timesteps should be the same for each installation right?

Comment (TeeeJay) [reply src/libecalc/presentation/exporter/dto/dtos.py:210]: not sure if it has or should, as I have experienced mismatches between mobile and fixed before ...

**Pull request merged**